File: react-app/src/components/NewComment/CommentForm.js
import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { createComment } from "../../store/comment";
import './commentForm.css';

const CommentForm = ({ post }) => {
    const dispatch = useDispatch();

    const user_id = useSelector((state) => state?.session?.user?.id);
    const [comment_body, setComment_body] = useState('');
    const [errors, setErrors] = useState([])


    const handleSubmit = async (e) => {
        e.preventDefault();

        const payload = {
            user_id,
            post_id: post?.id,
            comment_body
        }

        if (comment_body.length > 255) {
            return setErrors(['Comment must be 255 characters or less'])
        }


        const newComment = dispatch(createComment(payload));
        if (newComment) {
            setComment_body('');
            setErrors([]);
            return newComment;
        }
    }



    return (
        <div className="comment-form-container">
            {errors?.length > 0 && errors?.map((error) => <div key={error} className='comment-errors'>{error}</div>)}
            <form onSubmit={handleSubmit}>
                <input
                    className="comment-input"
                    placeholder="Add a comment..."
                    type='text'
                    value={comment_body}
                    onChange={(e) => setComment_body(e.target.value)}
                />
                <button className="comment-submit-button" type="submit">Post</button>
            </form>
        </div>
    );
}

export default CommentForm;

File: app/forms/comment_form.py
from flask_wtf import FlaskForm
from sqlalchemy import Integer
from wtforms import StringField, IntegerField, SubmitField
from wtforms.validators import DataRequired, ValidationError

def comment_body_length(form, field):
    comment_body = field.data
    if len(comment_body) > 255:
        raise ValidationError('Comment must be 255 characters or less.')

class CommentForm(FlaskForm):
    user_id = IntegerField('user_id')
    post_id = IntegerField('post_id')
    comment_body = StringField('comment_body', validators=[DataRequired(), comment_body_length])

    submit = SubmitField('submit')
